PWGCF/Flow/Tasks/resonancesGfwFlow.cxx

```diff
@@ -65,70 +64,136 @@ using namespace std;
 
 namespace
 {
-std::shared_ptr<TProfile> refC22Boot[10];
-std::shared_ptr<TProfile> refC24Boot[10];
-
-std::shared_ptr<TProfile3D> phiD22FPtBoot[10];
-std::shared_ptr<TProfile3D> phiD22BPtBoot[10];
-std::shared_ptr<TProfile3D> phiD24FPtBoot[10];
-std::shared_ptr<TProfile3D> phiD24BPtBoot[10];
-
-std::shared_ptr<TProfile3D> k0D22FPtBoot[10];
-std::shared_ptr<TProfile3D> k0D22BPtBoot[10];
-std::shared_ptr<TProfile3D> k0D24FPtBoot[10];
-std::shared_ptr<TProfile3D> k0D24BPtBoot[10];
-
-std::shared_ptr<TProfile3D> lambdaD22FPtBoot[10];
-std::shared_ptr<TProfile3D> lambdaD22BPtBoot[10];
-std::shared_ptr<TProfile3D> lambdaD24FPtBoot[10];
-std::shared_ptr<TProfile3D> lambdaD24BPtBoot[10];
-
-std::shared_ptr<TProfile3D> anLambdaD22FPtBoot[10];
-std::shared_ptr<TProfile3D> anLambdaD22BPtBoot[10];
-std::shared_ptr<TProfile3D> anLambdaD24FPtBoot[10];
-std::shared_ptr<TProfile3D> anLambdaD24BPtBoot[10];
+std::vector<std::shared_ptr<TProfile>> refV2;
+std::vector<std::shared_ptr<TProfile3D>> phiV2;
+std::vector<std::shared_ptr<TProfile3D>> k0V2;
+std::vector<std::shared_ptr<TProfile3D>> lambdaV2;
+
+std::vector<std::vector<std::shared_ptr<TProfile>>> refBoot;
+std::vector<std::vector<std::shared_ptr<TProfile3D>>> phiBoot;
+std::vector<std::vector<std::shared_ptr<TProfile3D>>> k0Boot;
+std::vector<std::vector<std::shared_ptr<TProfile3D>>> lambdaBoot;
 } // namespace
 
 #define O2_DEFINE_CONFIGURABLE(NAME, TYPE, DEFAULT, HELP) Configurable<TYPE> NAME{#NAME, DEFAULT, HELP};
 
+namespace o2::analysis::genericframework
+{
+GFWRegions regions;
+GFWCorrConfigs configs;
+} // namespace o2::analysis::genericframework
+
+template <typename T>
+auto projectMatrix(Array2D<T> const& mat, std::array<float, 6>& array1, std::array<float, 6>& array2, std::array<float, 6>& array3)
+{
+  for (auto j = 0; j < static_cast<int>(mat.cols); ++j) {
+    array1[j] = mat(0, j);
+    array2[j] = mat(1, j);
+    array3[j] = mat(2, j);
+  }
+  return;
+}
+template <typename T, typename P>
+auto readMatrix(Array2D<T> const& mat, P& array)
+{
+  for (auto i = 0; i < static_cast<int>(mat.rows); ++i) {
+    for (auto j = 0; j < static_cast<int>(mat.cols); ++j) {
+      array[i][j] = mat(i, j);
+    }
+  }
+
+  return;
+}
```

Comment:
Why it is needed to template on the kind of array? Isn't it completely determined and unique?